src/DynamoCore/Search/SearchElements/AutoCompletionNodeSearchElement.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dynamo.Configuration;
+using Dynamo.Graph.Nodes;
+
+namespace Dynamo.Search.SearchElements
+{
+    /// <summary>
+    ///      Class for all Auto-Completion Node search elements.
+    /// </summary>
+    public abstract class AutoCompletionNodeSearchElement
```

Comment:
Can you re-iterate the purpose of this class by comments here? We cant inherit the NodeSearchElement class because auto-completion API returns a list of NodeSearchElement and we would need this information in the NodeSearchElementViewModel to connect the port? I am fine with this, @mjkkirschner @aparajit-pratap what do you think? I think we plan to keep expanding here because we will have more properties coming in specific to AutoComplete function, e.g. user's rank, whether favored, ML rank